shpc/main/registry/__init__.py

```diff
@@ -171,18 +175,20 @@ def sync_from_remote(
 
         # No local registry provided, use default
         if not local:
-            local = Filesystem(self.settings.filesystem_registry)
+            local = self.filesystem_registry
```

Comment:
We will run into trouble if this is None (which it can be, will be by default!) It happens when they don't have a filesystem registry defined.

Reply:
We probably need to check for this, first thing, and cut out early if local isn't defined, or we don't have a filesystem registry (akin to sync above).